### PR TITLE
Add Account Abstraction (Biconomy) guide

### PR DESCRIPTION
**Pull request by taycaldwell**
**What changed? Why?**
Add a guide on implementing account abstraction into a Base project using Biconomy.

**Notes to reviewers**
N/A

**How has it been tested?**
Manually

**Does this PR add a new token to the bridge?**

- [x] No, this PR does not add a new token to the bridge
- [ ] I've confirmed this token doesn't use a bridge override
- [ ] I've confirmed this token is an OptimismMintableERC20

Please include evidence of both confirmations above for your reviewers.

### CODE REVIEW

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 1]
```diff
@@ -0,0 +1,830 @@
...
+
+:::
+
+---
+
+## Setting up the front-end
+
+### Creating a Next.js project
+
+After you set up your paymaster and bundler from the Biconomy Dashboard, the next step is to create a Next.js project for your app's front-end.
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:264]: change front-end to frontend

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 2]
```diff
@@ -0,0 +1,830 @@
...
+
+To use the paymaster and bundler that were setup from the Biconomy Dashboard, you will need to add a few dependencies to your Next.js project.
+
+To install Biconomy as a dependency to your project, run the following command:
+
+```bash
+yarn add @biconomy/account @biconomy/bundler @biconomy/common @biconomy/core-types @biconomy/paymaster ethers@5.7.2
+```
+
+Creating Biconomy smart accounts requires a signer from an EIP-1193 provider. Biconomy works with a variety of different social login and embedded wallet onboarding solutions that provide access to a signer that can be used for creating smart accounts. In this guide, you will use Particle Network for user authentication and getting a smart account signer.
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:283]: if EIP-1193 wasn't linked before, do so

Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:283]: link to define Particle Network

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 3]
```diff
@@ -0,0 +1,830 @@
...
+// highlight-next-line
+import { useState } from "react";
+
+export default function Home() {
+
+  // highlight-start
+  const [loading, setLoading] = useState(false);
+  const [provider, setProvider] = useState(null);
+  const [smartAccount, setSmartAccount] = useState(null);
+  const [address, setAddress] = useState("");
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:555]: use single quotes to be consistent?

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 4]
```diff
@@ -0,0 +1,830 @@
...
+
+```javascript
+"use client"
+
+import styles from "./page.module.css";
+import { ParticleAuthModule, ParticleProvider } from '@biconomy/particle-auth';
+// highlight-next-line
+import { ethers } from 'ethers';
+
+export default function Home() {
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:364]: Do we want to set these as constants at the top of the file?

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 5]
```diff
@@ -0,0 +1,830 @@
...
+
+// highlight-start
+import { IBundler, Bundler } from '@biconomy/bundler';
+import { IPaymaster, BiconomyPaymaster } from '@biconomy/paymaster';
+import { ChainId } from '@biconomy/core-types';
+import { DEFAULT_ENTRYPOINT_ADDRESS } from '@biconomy/account';
+// highlight-end
+
+export default function Home() {
+  const particle = new ParticleAuthModule.ParticleNetwork({
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:422]: Set as constants?

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 6]
```diff
@@ -0,0 +1,830 @@
...
+    clientKey: 'YOUR_PARTICLE_CLIENT_ID',
+    appId: 'YOUR_PARTICLE_APP_ID',
+    wallet: {
+      displayWalletEntry: true,
+    },
+  });
+
+  // highlight-start
+  const paymaster: IPaymaster = new BiconomyPaymaster({
+    paymasterUrl: 'YOUR_PAYMASTER_URL',
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:433]: Add a constant at the top

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 7]
```diff
@@ -0,0 +1,830 @@
...
+    const transactionDetails = await userOpResponse.wait();
+  } catch (e) {
+    console.error('Error executing transaction:', e);
+  }
+};
+```
+
+### Adding the contract ABI
+
+Initializing a contract instance requires the contracts ABI to be provided. The code for the Counter component in the previous section imports a file called `abi.json`:
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:729]: Spell out ABI the first time you use and add a link defining

**Review by rcbgr — DISMISSED** [review]

File: apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md [hunk 8]
```diff
@@ -0,0 +1,830 @@
...
+
+```bash
+cast wallet import deployer --interactive
+```
+
+After running the command above, you will be prompted to enter your private key, as well as a password for signing transactions.
+
+:::caution
+
+For instructions on how to get your private key from Coinbase Wallet, visit the [Coinbase Wallet documentation](https://docs.cloud.coinbase.com/wallet-sdk/docs/developer-settings#show-private-key). **It is critical that you do NOT commit this to a public repo.**
```
[the remark this answers is not in the document]
Comment (taycaldwell) [reply apps/base-docs/docs/building-with-base/guides/account-abstraction-with-biconomy.md:166]: Unlike using our private key in an `.env` file, for this specific command, the configuration is saved under a hidden directory outside of the project (i.e. `~/Users/<user>/.foundry/keystores/<wallet>`), so there is no risk of accidentally committing the file.

**Review by rcbgr — APPROVED** [review]

**Pull request merged**